src/LanguageServer/ProtocolUnitTests/Formatting/FormatDocumentTests.cs

```diff
@@ -46,7 +44,194 @@ void M()
+
+            var results = await RunFormatDocumentAsync(testLspServer, documentURI);
+            var actualText = ApplyTextEdits(results, documentText);
+            AssertEx.EqualOrDiff(expected, actualText);
+        }
+
+        [Theory, CombinatorialData]
+        public async Task TestFormatDocumentWithOrganizeImports3Async(bool mutatingLspWorkspace)
+        {
+            var markup =
+                """
+                using  System;
+                using   System.Collections;
+
+                class A{|caret:|}
+                {
+                }
+                """;
+            var expected =
+                """
+                using System;
+                using System.Collections;
+
+                class A
+                {
+                }
+                """;
+
+            var options = new InitializationOptions
+            {
+                OptionUpdater = globalOptions =>
+                {
+                    globalOptions.SetGlobalOption(LspOptionsStorage.LspOrganizeImportsOnFormat, LanguageNames.CSharp, true);
+                },
+            };
+
+            await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace, options);
+            var documentURI = testLspServer.GetLocations("caret").Single().Uri;
+            var documentText = await testLspServer.GetDocumentTextAsync(documentURI);
+
+            var results = await RunFormatDocumentAsync(testLspServer, documentURI);
+            var actualText = ApplyTextEdits(results, documentText);
+            AssertEx.EqualOrDiff(expected, actualText);
+        }
+
+        [Theory, CombinatorialData]
+        public async Task TestFormatDocumentWithOrganizeImports4Async(bool mutatingLspWorkspace)
+        {
+            var markup =
+                """
+                using System;
+                using System.Collections;
+
+                class  A{|caret:|}
+                {
+                }
+                """;
+            var expected =
+                """
+                using System;
+                using System.Collections;
+
+                class A
+                {
+                }
+                """;
+
+            var options = new InitializationOptions
+            {
+                OptionUpdater = globalOptions =>
+                {
+                    globalOptions.SetGlobalOption(LspOptionsStorage.LspOrganizeImportsOnFormat, LanguageNames.CSharp, true);
+                },
+            };
+
+            await using var testLspServer = await CreateTestLspServerAsync(markup, mutatingLspWorkspace, options);
+            var documentURI = testLspServer.GetLocations("caret").Single().Uri;
+            var documentText = await testLspServer.GetDocumentTextAsync(documentURI);
+
+            var results = await RunFormatDocumentAsync(testLspServer, documentURI);
+            var actualText = ApplyTextEdits(results, documentText);
+            AssertEx.EqualOrDiff(expected, actualText);
```

Comment:
test where both the class *and* teh imports need formatting *and* sorting changes

Reply:
Added!